addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -12,178 +13,223 @@
+
+// Remove invalid/non-existent entries
+_grenadeList = _grenadeList - [""];
+_putList = _putList - [""];
 
+// Get all other grenades, explosives (and similar) and magazines
 {
-    private _className = configName _x;
+    _className = configName _x;
 
-    switch true do {
+    switch (true) do {
         // Grenades
         case (_className in _grenadeList): {
-            (_cargo select 15) pushBackUnique _className;
+            (_configItems select IDX_VIRT_GRENADES) pushBackUnique _className;
         };
-        // Put
+        // Explosives
         case (_className in _putList): {
-            (_cargo select 16) pushBackUnique _className;
+            (_configItems select IDX_VIRT_EXPLOSIVES) pushBackUnique _className;
         };
-        // Rifle, handgun, secondary weapons mags
+        // Primary, handgun & secondary weapon magazines, and magazines that are forced with "ace_arsenal_hide = -1"
         case (
-                ((getNumber (_x >> "type") in [TYPE_MAGAZINE_PRIMARY_AND_THROW,TYPE_MAGAZINE_SECONDARY_AND_PUT,1536,TYPE_MAGAZINE_HANDGUN_AND_GL,TYPE_MAGAZINE_MISSILE]) ||
-                {(getNumber (_x >> QGVAR(hide))) == -1})
-            ): {
-            (_cargo select 2) pushBackUnique _className;
+            getNumber (_x >> QGVAR(hide)) == -1 ||
+            {getNumber (_x >> "type") in [TYPE_MAGAZINE_PRIMARY_AND_THROW, TYPE_MAGAZINE_SECONDARY_AND_PUT, 1536, TYPE_MAGAZINE_HANDGUN_AND_GL, TYPE_MAGAZINE_MISSILE]}
+        ): {
+            (_configItems select IDX_VIRT_ITEMS_ALL) pushBackUnique _className;
         };
     };
-} foreach configProperties [(configFile >> "CfgMagazines"), "isClass _x && {(if (isNumber (_x >> 'scopeArsenal')) then {getNumber (_x >> 'scopeArsenal')} else {getNumber (_x >> 'scope')}) == 2} && {getNumber (_x >> 'ace_arsenal_hide') != 1}", true];
+} forEach configProperties [_cfgMagazines, _filterFunction, true];
 
+// Get all backpacks
 {
     if (getNumber (_x >> "isBackpack") == 1) then {
-        (_cargo select 6) pushBackUnique (configName _x);
+        (_configItems select IDX_VIRT_BACKPACK) pushBackUnique (configName _x);
     };
-} foreach configProperties [(configFile >> "CfgVehicles"), "isClass _x && {(if (isNumber (_x >> 'scopeArsenal')) then {getNumber (_x >> 'scopeArsenal')} else {getNumber (_x >> 'scope')}) == 2} && {getNumber (_x >> 'ace_arsenal_hide') != 1}", true];
+} forEach configProperties [configFile >> "CfgVehicles", _filterFunction, true];
 
+// Get all facewear
 {
-    (_cargo select 7) pushBackUnique (configName _x);
-} foreach configProperties [(configFile >> "CfgGlasses"), "isClass _x && {(if (isNumber (_x >> 'scopeArsenal')) then {getNumber (_x >> 'scopeArsenal')} else {getNumber (_x >> 'scope')}) == 2} && {getNumber (_x >> 'ace_arsenal_hide') != 1}", true];
+    (_configItems select IDX_VIRT_GOGGLES) pushBackUnique (configName _x);
+} forEach configProperties [configFile >> "CfgGlasses", _filterFunction, true];
 
+// Get magazine groups (magazinewells)
```

Comment:
The magazine groups are no longer used by the arsenal. Do we still keep the code?

Reply:
If it really isn't used/needed anymore. I'm inclined to say no.
Other mods may be piggy-bagging on our code and use our scanned values from our cache.
But these are not marked as public/api so mods shouldn't.